File: lib/core/ai/gemini_provider.dart
import 'dart:async';
import 'ai_provider.dart';

/// GeminiProvider stub. Keeps parity with other providers to make swapping
/// providers trivial later on.
class GeminiProvider implements AIProvider {
  @override
  String get providerId => 'gemini';

  @override
  String get providerName => 'Gemini (stub)';

  @override
  String get description => 'Gemini provider (stub) — implement actual Gemini calls here.';

  @override
  String get assistantName => 'Dharma';

  @override
  bool get supportsStreaming => false;

  @override
  Future<String> sendMessage(String prompt, {Map<String, dynamic>? options}) async {
    await Future.delayed(const Duration(milliseconds: 380));
    return '[GeminiStub] Dharma: Response available.';
  }

  @override
  Stream<String> streamMessage(String prompt, {Map<String, dynamic>? options}) async* {
    yield await sendMessage(prompt, options: options);
  }

  @override
  Future<void> cancel() async {}
}

File: server/src/providers/geminiProvider.ts
import { EventEmitter } from 'events';

class GeminiProvider {
  providerId = 'gemini';

  async sendMessage(_prompt: string, _opts?: any): Promise<string> {
    await new Promise((r) => setTimeout(r, 420));
    return '[Gemini Stub] Dharma: response';
  }

  streamMessage(prompt: string, _opts?: any): EventEmitter {
    const e = new EventEmitter();
    const text = `[Gemini Stub streaming] ${prompt}`;
    const parts = text.split(' ');
    let i = 0;
    const t = setInterval(() => {
      if (i >= parts.length) { clearInterval(t); e.emit('end'); } else { e.emit('data', parts[i] + (i === parts.length - 1 ? '' : ' ')); }
      i++;
    }, 90);
    return e;
  }

  async cancel() {}
}

export default GeminiProvider;
